lib/zxystream.js

```diff
@@ -0,0 +1,57 @@
+var util = require('util');
+var split = require('split');
+var s3scan = require('s3scan');
+var s3urls = require('s3urls');
+var StreamConcat = require('stream-concat');
+
+module.exports = ZXYStream;
+module.exports.config = config;
+
+function ZXYStream(tilesUrl) {
+    var conf = config(tilesUrl);
+    var current = 0;
+    return new StreamConcat(function() {
+        if (!conf.toList[current]) return null;
+        return s3scan.List(conf.toList[current++], {})
+    }).pipe(split(function(line) {
+        var matches = line.match(conf.tokenPattern);
+        if (!matches || matches.length < 4) return undefined;
+        return matches[conf.tokenMap[0]] + '/' +
+            matches[conf.tokenMap[1]] + '/' +
+            matches[conf.tokenMap[2]] + '\n';
+    }));
+}
+
+function config(tilesUrl) {
+    var conf = {};
+    conf.toList = [];
+    if ((/{prefix}/).test(tilesUrl)) {
```

Comment:
I have a branch of s3scan lurking somewhere that could navigate 2-char hex `{prefix}`s in urls. Would you prefer to keep the logic here in tilelive-s3, or do you think it would be broadly useful there?